tasks/lib/compiler.js

```diff
@@ -47,7 +48,7 @@ var Compiler = function(grunt, options, cwd) {
     // Append formatted URL
     path += Url.format( Url.parse( url.replace(/\\/g, '/') ) );
 
-    return "\n  $templateCache.put('" + path + "',\n    " + template + "\n  );\n";
+    return options.templateWrap(path, template, index, files);
```

Comment:
Can you give a reason why the template wrapper function might need the index or the full list of files? It doesn't look like they're used in your examples.